portal/src/graphql/portal/SingleSignOnConfigurationScreen.tsx

```diff
@@ -124,6 +124,11 @@ function getOrCreateSecret(secretConfigState: PortalAPISecretConfig) {
       data: { items: [] },
     };
     secretConfigState.secrets.push(secret);
+    // Sync secret item order with server side.
+    // Otherwise the error json pointer would not match.
+    secretConfigState.secrets = secretConfigState.secrets.sort((a, b) =>
```

Comment:
Why do we need this sorting? the json pointer should refer to the new secret config sent from portal, so we just need to find the index with secretConfigState and it should always sync with server side no matter what order `secretConfigState.secrets` is in?

Reply:
In server side, the secret items are sorted by their key.